v2/api/dbformysql/customizations/flexible_server_extensions.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package customizations
+
+import (
+	"context"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	mysql "github.com/Azure/azure-service-operator/v2/api/dbformysql/v1beta20210501storage"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	. "github.com/Azure/azure-service-operator/v2/internal/logging"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/secrets"
+)
+
+var _ extensions.SecretsRetriever = &FlexibleServerExtension{}
+
+func (ext *FlexibleServerExtension) RetrieveSecrets(
+	ctx context.Context,
+	obj genruntime.ARMMetaObject,
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger) ([]*v1.Secret, error) {
+
+	// This has to be the current storage version. It will need to be updated
+	// if the storage version changes.
```

Comment:
Suggest being very explicit to avoid confusion with our internal storage version concept. 

```suggestion
	// This has to be the current hub storage version. It will need to be updated
	// if the hub storage version changes.
```